### ISSUE TITLE
wrong result :test_expr('len("123"+456)', true) 

### ISSUE THREAD

**Issue by nhmice** [foo123/Xpresion#1]
test_expr('len("123"+456)', true) 
-------------------------------------
Expression: len("123"+456)
Variables : []
Evaluator : $Fn->len(("123"+456))
Data      : array(0) {
}
Result    : int(1)----wrong

test_expr('len("123"+"456")', true) 
-------------------------------------
Expression: len("123"+"456")
Variables : []
Evaluator : $Fn->len(("123"+"456"))
Data      : array(0) {
}
Result    : int(1)----wrong

test_expr('len(("123"+"456"))', true) 
-------------------------------------
Expression: len( ("123"+"456"))
Variables : []
Evaluator : $Fn->len(("123"+"456"))
Data      : array(0) {
}
Result    : int(1)----wrong
-------------------------------------
This error may be a function parameter analysis error. Expression ” Len (456) “ was executed, so the default value of 1 was returned。

-------------------------------------
Expression: sum(1,2+3,4)
Variables : []
Evaluator : $Fn->sum((1,2+3,4))
Data      : array(0) {
}
Result    : NULL------wrong
-------------------------------------
Expression: sum(1,(2+3),4)
Variables : []
Evaluator : $Fn->sum(1,(2+3),4)
Data      : array(0) {
}
Result    : int(10)-----right
-------------------------------------

**Comment by foo123** [comment]
Thanks looking into it. If you have more bugs let me know to try and fix them all before uploading new version

**Comment by foo123** [comment]
Fixed in version 1.0.1 along with a couple of other minor issues, see associated tests.

For example (for PHP, https://github.com/foo123/Xpresion/blob/master/test/out-php#L3 ):

```text
Xpresion.VERSION 1.0.1

==========================================
Expression: (123+456)
Variables : []
Evaluator : (123+456)
Data      : array(0) {
}

Result    : int(579)

==========================================
Expression: ("123"+456)
Variables : []
Evaluator : ("123".strval(456))
Data      : array(0) {
}

Result    : string(6) "123456"

==========================================
Expression: ("123"+"456")
Variables : []
Evaluator : ("123".strval("456"))
Data      : array(0) {
}

Result    : string(6) "123456"

==========================================
Expression: len("123"+456)
Variables : []
Evaluator : $Fn->len(("123".strval(456)))
Data      : array(0) {
}

Result    : int(6)

==========================================
Expression: len("123"+"456")
Variables : []
Evaluator : $Fn->len(("123".strval("456")))
Data      : array(0) {
}

Result    : int(6)
==========================================
Expression: sum(1,2+3,4)
Variables : []
Evaluator : $Fn->sum(1,(2+3),4)
Data      : array(0) {
}

Result    : int(10)

==========================================
Expression: sum(1,(2+3),4)
Variables : []
Evaluator : $Fn->sum(1,(2+3),4)
Data      : array(0) {
}

Result    : int(10)
==========================================
Expression: "1,2,3"+3+len("123")
Variables : []
Evaluator : (("1,2,3".strval(3)).strval($Fn->len("123")))
Data      : array(0) {
}

Result    : string(7) "1,2,333"

==========================================
Expression: "1,2,3"+3+sum(1,2,3)
Variables : []
Evaluator : (("1,2,3".strval(3)).strval($Fn->sum(1,2,3)))
Data      : array(0) {
}

Result    : string(7) "1,2,336"

==========================================
Expression: sum(1,len("123"+456),2)
Variables : []
Evaluator : $Fn->sum(1,$Fn->len(("123".strval(456))),2)
Data      : array(0) {
}

Result    : int(9)
```

Closing this issue, if you find other bugs open a new one. Cheers!
